common/dev/xdpe12284c.c

```diff
@@ -158,6 +193,391 @@ static float vid_to_float(int val, uint8_t vout_mode)
 	return 0;
 }
 
+static bool parsing_image(uint8_t *hex_buff, struct xdpe_config *dev_cfg)
+{
+	CHECK_NULL_ARG_WITH_RETURN(hex_buff, false);
+	CHECK_NULL_ARG_WITH_RETURN(dev_cfg, false);
+
+	bool ret = false;
+	int max_data_size = MAX_CMD_LINE;
+	dev_cfg->data = (uint8_t *)malloc(max_data_size);
+	if (!dev_cfg->data) {
+		LOG_ERR("Failed to malloc data!");
+		goto exit;
+	}
+
+	dev_cfg->crc_exp = 0;
+
+	bool rec_flag = false;
+	bool new_line = true;
+	bool got_end_flag = false;
+	uint16_t ofst = 0;
+	uint16_t value = 0;
+	int data_idx = 0, val;
+	for (int i = 0; i < fw_update_cfg.image_size; i++) {
+		/* check valid */
+		if (!hex_buff[i]) {
+			LOG_ERR("Get invalid buffer data at index %d", i);
+			goto exit;
+		}
+
+		/* collect data */
+		if (rec_flag == true) {
```

Comment:
Is there a way to simplify this set of if statements.  Going 5 levels deep is too many to follow and maintain easily.